### DISCUSSION TITLE
Two way authentication / keyboard_interactive method

### DISCUSSION THREAD

**Discussion by mimoshe** [ParallelSSH/parallel-ssh#345] [Q&A] [answered]
Hello team, 
I need to connect to Linux server that uses two way authentication per user
for example in putty when I ssh it will ask for username, then prompt for PIN once this provided, it will prompt for a password, once that provided the server will accept the handshake and create the connection.
Any idea how accomplish this with ssh2-Python?

so far my code is simple

`import socket
from ssh2.session import Session
from getpass import getpass

host = input("Enter Hostname :")
user = input("Enter Username :")
password = getpass("Enter Password :")

sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
sock.connect((host, 22))
session = Session()

session.handshake(sock)
session.userauth_keyboardinteractive(user, password)

channel = session.open_session()
channel.execute('ls -l')
size, data = channel.read()
while size > 0:
    print(data.decode())
    size, data = channel.read()

channel.close()
print("Exit status: {0}".format(channel.get_exit_status()))`

currently the [session.userauth_keyboardinteractive(user, password)] line is getting ssh2.exceptions.AuthenticationError

Any help is appreciated

**Comment by pkittenis** [comment, marked as answer]
Try with the [standard username/password authentication feature ](https://parallel-ssh.readthedocs.io/en/latest/quickstart.html#user-password-authentication) and see if that works for you.
